### DISCUSSION TITLE
what is wherecondition pattern

### DISCUSSION THREAD

**Discussion by da-da-da** [nocodb/nocodb#7196] [Q&A] [answered]
hi, I'm adding some where condition to return data by user's name or anything value in column. But I'm not sure how to joint condition or the where condition pattern in `param.query.where`. When I read BaseModelSqlv2.extractFilterFromXwhere, I find this function maybe support more than one where condition like `(Creator,like,%${userEmail}%) and (name,like,%${username}%)` or `(~and(~or(Title,like,%xx%),~or(Author,eq,yy)))`. Please help me to get what is the correct pattern to joint multiple where condition.

**Comment by pranavxc** [comment, marked as answer]
You can refer following documentation - https://docs.nocodb.com/0.109.7/developer-resources/rest-apis#logical-operators

for example:

```
(Title,like,%xx%)~and(Author,eq,yy)

// equivalent to `(tiltle like '%xx%') AND (Author = 'yy')`
```

**Reply by da-da-da** [reply]
thank you so much. I'll try this😙
